hack/tests/subcommand.sh

```diff
@@ -30,14 +30,28 @@ kubectl create namespace test-memcached
 operator-sdk test local ./test/e2e --up-local --namespace=test-memcached --kubeconfig $KUBECONFIG
 kubectl delete namespace test-memcached
 
+# test operator in up local mode with namespace set to ""
+# --namespace="" is allowed in local mode only when --no-setup is set
+kubectl create namespace test-memcached
+kubectl create -f deploy/crds/cache.example.com_memcacheds_crd.yaml
+# this runs after the popd at the end (trap_add popd), so it needs the path from the project root
+trap_add 'kubectl delete --ignore-not-found test/test-framework/deploy/crds/cache.example.com_memcacheds_crd.yaml' EXIT
+kubectl create -f deploy/crds/cache.example.com_memcachedrs_crd.yaml
+# this runs after the popd at the end (trap_add popd), so it needs the path from the project root
+trap_add 'kubectl delete --ignore-not-found test/test-framework/deploy/crds/cache.example.com_memcachedrs_crd.yaml' EXIT
+operator-sdk test local ./test/e2e --up-local --namespace="" --no-setup --debug
```

Comment:
Why the debug is required?
Why the resources are not applied in the namespace created?

Reply:
> Why you are using the --no-setup? Should not it work with only operator-sdk test local ./test/e2e --up-local --namespace=""?

because, in the current setup, `--namespace=""` is allowed without `--no-setup` expectations/testts can be broken. 
by requiring `--no-setup` we can ensure that a developer is responsible for the setup of namespaces and global resources before the `--namespace=""` tests

Reply:
we can close this pr as pr https://github.com/operator-framework/operator-sdk/pull/2149
address these better